Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/BaseKeyPackageSyncTool.kt

```diff
@@ -87,19 +87,20 @@ open class BaseKeyPackageSyncTool(
         return@filter true
     }
 
-    internal suspend fun getDownloadedCachedKeys(
+    internal suspend fun getCachedKeys(
         location: LocationCode,
         type: CachedKeyInfo.Type
     ): List<CachedKey> = keyCache.getEntriesForType(type)
         .filter { it.info.location == location }
         .filter { key ->
             val complete = key.info.isDownloadComplete
             val exists = key.path.exists()
-            if (complete && !exists) {
+            val checked = key.info.checkedForExposures
+            if (complete && !exists && !checked) {
                 Timber.tag(tag).v("Incomplete download, will overwrite: %s", key)
             }
-            // We overwrite not completed ones
-            complete && exists
+            // We overwrite incomplete ones that have not been checked
+            complete && (exists || checked)
```

Comment:
These 4 lines are confusing, in `if` checks for `!` existing `!` checked keys and logs call them incomplete:
```kotlin
if (complete && !exists && !checked) {
   Timber.tag(tag).v("Incomplete download, will overwrite: %s", key)
}
```
BUT in here, we return complete and (existing or checked)  and the comment says incomplete too:
```kotlin 
// We overwrite incomplete ones that have not been checked
complete && (exists || checked)
```
Could you please elaborate more ? or correct the log / comment statements?